src/librbd/ManagedLock.h

```diff
@@ -0,0 +1,184 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_LIBRBD_MANAGED_LOCK_H
+#define CEPH_LIBRBD_MANAGED_LOCK_H
+
+#include "include/int_types.h"
+#include "include/Context.h"
+#include "include/rados/librados.hpp"
+#include "cls/lock/cls_lock_types.h"
+#include "common/Mutex.h"
+#include <list>
+#include <string>
+#include <utility>
+
+class ContextWQ;
+
+namespace librbd {
+
+template <typename Watcher>
```

Comment:
I think I'm not following what you said. The `librbd::Watcher` class is not templated and therefore I cannot create a specialization in the unit tests.
Are you suggesting to also template the `Watcher` class on `ImageCtx` to easily create mocks of the Watcher class?